src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -581,4 +581,28 @@ object messages {
     }
   }
 
+
+  case class ByNameParameterNotSupported()(implicit ctx: Context) extends Message(21) {
+    val kind = "Syntax"
+
+    val msg = "By-name parameter type not allowed here."
+
+    val explanation =
+      hl"""|By-name parameters act like functions that are only evaluated when referenced,
+           |allowing for lazy evaluation of a parameter.
+           |
+           |An example of using a by-name parameter would look like:
+           |${"def func(f: => Boolean) = f"} // 'f' is evaluated when it is referenced within the function
```

Comment:
I'd move the `}` to the end of the line so the full line is highlighted (and I'd cut out "it is" so that the line doesn't exceed 80 columns)